src/RNTuple/Writing/TFileWriter.jl

```diff
@@ -484,12 +484,16 @@ end
 
 # vector case
 function add_field_column_record!(field_records, column_records, input_T::Type{<:AbstractVector}, NAME; parent_field_id, col_field_id = parent_field_id)
+    implicit_field_id = length(field_records)
     fr =  UnROOT.FieldRecord(; field_version=0x00000000, type_version=0x00000000, parent_field_id, struct_role=0x0001, flags=0x0000, repetition=0, source_field_id=-1, root_streamer_checksum=-1, field_name=string(NAME), type_name="", type_alias="", field_desc="", )
     push!(field_records, fr)
-    cr_offset = UnROOT.ColumnRecord(RNTUPLE_WRITE_TYPE_IDX_DICT[Index64]..., parent_field_id, 0x00, 0x00, 0)
+    cr_offset = UnROOT.ColumnRecord(RNTUPLE_WRITE_TYPE_IDX_DICT[Index64]..., col_field_id, 0x00, 0x00, 0)
     push!(column_records, cr_offset)
 
-    add_field_column_record!(field_records, column_records, eltype(input_T), "_0"; parent_field_id = parent_field_id, col_field_id = parent_field_id+1)
+    # TODO: this feels like a hack, think about it more
+    Element_T = eltype(input_T)
+    content_parent_field_id = Element_T <: Real ? implicit_field_id : parent_field_id
+    add_field_column_record!(field_records, column_records, Element_T, "_0"; parent_field_id = content_parent_field_id, col_field_id = length(field_records))
```

Comment:
this is really not elegant and I feel like this would fail in more complicated case (i.e. not sure what happens when you have structs or 4-th order jagging)